streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -552,28 +554,27 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
         return doJoin(other, joiner, windows, joined, new KStreamImplJoin(true, true));
     }
 
-    private <V1, R> KStream<K, R> doJoin(final KStream<K, V1> other,
-                                         final ValueJoiner<? super V, ? super V1, ? extends R> joiner,
-                                         final JoinWindows windows,
-                                         final Joined<K, V, V1> joined,
-                                         final KStreamImplJoin join) {
+    private <VO, VR> KStream<K, VR> doJoin(final KStream<K, VO> other,
```

Comment:
Renamed these to match the other names used in this class

Reply:
Should the `<V1, R> KStream<K, R> outerJoin` and `<V1, R> KStream<K, R> join` above be changed as well?

Reply:
In response to this comment, I've gone through the whole `KStreamImpl` and standardized the generic types in the following way:
* `K,V` (unchanged) the key/value types of `this` stream
* `KR/VR` the "result" key/value types where applicable
* `KO/VO` the "other" key/value types (where there's an "other" stream involved)
* `KG/VG` the key/value types when there's a global table involved
* `K1/V1` key/value types with no particular semantics that are scoped to the method (I.e., not the same as the instance-level `this` `K/V`)